ldap.go

```diff
@@ -271,14 +271,16 @@ func addRequestDescriptions(packet *ber.Packet) error {
 
 func addDefaultLDAPResponseDescriptions(packet *ber.Packet) error {
 	err := GetLDAPError(packet)
-	packet.Children[1].Children[0].Description = "Result Code (" + LDAPResultCodeMap[err.(*Error).ResultCode] + ")"
-	packet.Children[1].Children[1].Description = "Matched DN (" + err.(*Error).MatchedDN + ")"
-	packet.Children[1].Children[2].Description = "Error Message"
-	if len(packet.Children[1].Children) > 3 {
-		packet.Children[1].Children[3].Description = "Referral"
-	}
-	if len(packet.Children) == 3 {
-		return addControlDescriptions(packet.Children[2])
+	if err != nil {
```

Comment:
There's a subtle issue here, partially introduced by PR #191 where the `getLDAPResultCode` function was replaced by `GetLDAPError` - the semantics changed subtly with this PR, and I missed the implications when I merged it.

Even if there is no error, the original intent of this function is to add `Descriptions`; I think this may be a more useful change:

```
diff --git ldap.go ldap.go
index d766667..831fa02 100644
--- ldap.go
+++ ldap.go
@@ -270,10 +270,13 @@ func addRequestDescriptions(packet *ber.Packet) error {
 }
 
 func addDefaultLDAPResponseDescriptions(packet *ber.Packet) error {
-       err := GetLDAPError(packet)
-       packet.Children[1].Children[0].Description = "Result Code (" + LDAPResultCodeMap[err.(*Error).ResultCode] + ")"
-       packet.Children[1].Children[1].Description = "Matched DN (" + err.(*Error).MatchedDN + ")"
-       packet.Children[1].Children[2].Description = "Error Message"
+       resultCode := uint16(LDAPResultSuccess)
+       if err := GetLDAPError(packet); err != nil {
+               resultCode = err.(*Error).ResultCode
+               packet.Children[1].Children[2].Description = "Error Message"
+       }
+       packet.Children[1].Children[0].Description = "Result Code (" + LDAPResultCodeMap[resultCode] + ")"
+       packet.Children[1].Children[1].Description = "Matched DN"
        if len(packet.Children[1].Children) > 3 {
                packet.Children[1].Children[3].Description = "Referral"
        }

```


I may need to find a good way to back out the previous PR that skews the original intent.

Reply:
Any news here? I'm affected by this too. I don't mind to make another PR trying to address it and other possible occurrences of this same problem (If I find them).